components/datetime/src/skeleton.rs

```diff
@@ -348,12 +349,169 @@ const REQUESTED_SYMBOL_MISSING: u32 = 10000;
 /// there is no attempt to add on missing fields. This enum encodes the variants for the current
 /// search for a best skeleton.
 #[derive(Debug, PartialEq, Clone)]
-pub enum BestSkeleton<'a> {
-    AllFieldsMatch(AvailableFormatPattern<'a>),
-    MissingOrExtraFields(AvailableFormatPattern<'a>),
+pub enum BestSkeleton<T> {
+    AllFieldsMatch(T),
+    MissingOrExtraFields(T),
     NoMatch,
 }
 
+// TODO - This could return a Cow<'a, Pattern>, but it affects every other part of the API to
```

Comment:
Hopefully Zibi will be able to fix this in his PR to make it such that patterns in DateTimeFormat can always be zero-copy.